src/dashboard/utilities.py

```diff
@@ -46,3 +47,72 @@ def toggle_classname(class_str: str, class_to_toggle: str) -> str:
         return set_classname(class_str, class_to_toggle, False)
     else:
         return set_classname(class_str, class_to_toggle, True)
+
+
+def pluralize(unit: str, amount: int) -> str:
+    """Pluralize a unit if the amount is greater than 1.
+
+    Args:
+        unit (str): The unit.
+        amount (int): The amount.
+
+    Returns:
+        str: The unit with s appended
+        if amount is greater than 1,
+        otherwise the unit.
+    """
+    if amount == 1:
+        return unit
+
+    return f"{unit}s"
+
+
+def singularize(article: str, amount: int) -> str:
+    """Singularize an amount if it is equal to one.
+
+    Args:
+        article (str): The indefinite
+        article of an amount of something.
+        E.g. "A" or "An" in english.
+        amount (int): An amount.
+
+    Returns:
+        str: `article` if the amount
+        is equal to 1, otherwise
+        a string of amount.
+    """
+    if amount == 1:
+        return article
+
+    return str(amount)
+
+
+def to_human_time_delta(t1: datetime, t2: datetime) -> str:
+    """Convert a time delta to a human readable form.
+
+    Args:
+        t1 (datetime): A point in time.
+        t2 (datetime): A point in time.
+
+    Returns:
+        str: A human readable form of
+        the delta.
+    """
+    delta = abs(t1 - t2)
+    units: list[tuple[str, str, int]] = [
+        ("A", "week", 604800),
+        ("A", "day", 86400),
+        ("An", "hour", 3600),
+        ("A", "minute", 60),
+        ("A", "second", 1),
```

Comment:
Using timedelta for the duration lengths here might be more readable and less error-prone. The durations in seconds seem like magic numbers to me.
```python
[
    ("A", "month", timedelta(days=30)),
    ("A", "week", timedelta(weeks=1),
    ...
]
```

We could also use abbreviated units if the long version clutters the website: s, m, h, d, w, mo, y.

Reply:
Using timedelta sounds good. Could the abbreviated units only be displayed if passed as an optional parameter to the function?